Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs

```diff
@@ -296,31 +312,44 @@ private CSProjectInfo GetProjectInfo(Dictionary<string, CSProjectInfo> projectsM
             return toReturn;
         }
 
-        private List<PluginAssemblyInfo> ScanForPluginDLLs()
+        private void ScanForReferences(out List<PluginAssemblyInfo> plugins, out List<WinMDInfo> winmds)
         {
-            List<PluginAssemblyInfo> toReturn = new List<PluginAssemblyInfo>();
+            plugins = new List<PluginAssemblyInfo>();
+            winmds = new List<WinMDInfo>();
 
-            foreach (string assetAssemblyPath in Directory.GetFiles(Utilities.AssetPath, "*.dll", SearchOption.AllDirectories))
+            IEnumerable<string> assetReferences = Directory.GetFiles(Utilities.AssetPath, "*.*", SearchOption.AllDirectories)
```

Comment:
Could you use EnumerateFiles? As this would otherwise return a massive list of files.